### ISSUE TITLE
`lintr::missing_argument_linter` should exclude `magrittr::extract()` by default

### ISSUE THREAD

**Issue by salim-b** [r-lib/lintr#2109]
Consider the following reprex:

``` r
mtcars |> magrittr::extract(1)
#>                      mpg
#> Mazda RX4           21.0
#> Mazda RX4 Wag       21.0
#> Datsun 710          22.8
#> Hornet 4 Drive      21.4
#> Hornet Sportabout   18.7
#> Valiant             18.1
#> Duster 360          14.3
#> Merc 240D           24.4
#> Merc 230            22.8
#> Merc 280            19.2
#> Merc 280C           17.8
#> Merc 450SE          16.4
#> Merc 450SL          17.3
#> Merc 450SLC         15.2
#> Cadillac Fleetwood  10.4
#> Lincoln Continental 10.4
#> Chrysler Imperial   14.7
#> Fiat 128            32.4
#> Honda Civic         30.4
#> Toyota Corolla      33.9
#> Toyota Corona       21.5
#> Dodge Challenger    15.5
#> AMC Javelin         15.2
#> Camaro Z28          13.3
#> Pontiac Firebird    19.2
#> Fiat X1-9           27.3
#> Porsche 914-2       26.0
#> Lotus Europa        30.4
#> Ford Pantera L      15.8
#> Ferrari Dino        19.7
#> Maserati Bora       15.0
#> Volvo 142E          21.4
mtcars |> magrittr::extract(1, 1)
#> [1] 21
mtcars |> magrittr::extract(1, )
#>           mpg cyl disp  hp drat   wt  qsec vs am gear carb
#> Mazda RX4  21   6  160 110  3.9 2.62 16.46  0  1    4    4

# lintr complains about last expr from above
lintr::lint(text = 'mtcars |> magrittr::extract(1, )',
            linters = lintr::missing_argument_linter())
#> <text>:1:30: style: [missing_argument_linter] Missing argument in function call.
#> mtcars |> magrittr::extract(1, )
#>                              ^
```

<sup>Created on 2023-09-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

The lintr complaint would be applicable to `magrittr::extract2()` (e.g. `mtcars |> magrittr::extract2(1, )`), but not `magrittr::extract()`.

Could `magrittr::extract` be added to the default `except` value of `lintr::missing_argument_linter()`?

**Comment by MichaelChirico** [comment]
Hmm, I _thought_ we explicitly decided to leave packages out of the `except=` argument for `missing_argument_linter()`, but can't track down where that was written if so. Specifically since `rlang::list2()` would be a good general candidate, but then tons of tidyverse functions that build on `list2()` would be too.

So I think it's best to leave it up to the maintainer to write out their own `except=` as needed. I'm particularly wary of adding `extract()` which could come from anywhere, but we don't have a good way anywhere in the package yet of ensuring `foo::extract()` is skipped but `magrittr::extract()` is linted.
